src/shared-components/button/index.tsx

```diff
@@ -63,99 +63,83 @@ type ButtonProps = {
+  buttonType = 'primary',
+  children,
+  disabled = false,
+  icon = null,
+  isFullWidth = false,
+  isLoading = false,
+  loading = undefined,
+  onClick = () => undefined,
+  textColor = '',
+  ...rest
+}: ButtonProps) => {
+  const loadingVal = loading === undefined ? isLoading : loading;
 
-  static propTypes = {
-    buttonColor: COLORS_PROP_TYPES,
-    buttonType: PropTypes.oneOf([
-      'primary',
-      'secondary',
-      'tertiary',
-      'quaternary',
-    ]),
-    children: PropTypes.node.isRequired,
-    disabled: PropTypes.bool,
-    icon: PropTypes.node,
-    isFullWidth: PropTypes.bool,
-    isLoading: PropTypes.bool,
-    loading: isLoadingPropFunction,
-    onClick: PropTypes.func,
-    textColor: PropTypes.string,
-  };
-
-  static defaultProps = {
-    buttonColor: COLORS.primary,
-    buttonType: 'primary',
-    disabled: false,
-    icon: null,
-    isFullWidth: false,
-    isLoading: false,
-    loading: undefined,
-    onClick: () => undefined,
-    textColor: '',
-  };
-
-  render() {
-    const {
-      buttonColor = COLORS.primary,
-      buttonType = 'primary',
-      children,
-      disabled = false,
-      icon = null,
-      isFullWidth = false,
-      isLoading = false,
-      loading = undefined,
-      onClick = () => undefined,
-      textColor = '',
-      ...rest
-    } = this.props;
-
-    const loadingVal = loading === undefined ? isLoading : loading;
-
-    return (
-      <ButtonBase
-        disabled={disabled}
-        onClick={
-          !disabled && !loadingVal ? onClick : (event) => event.preventDefault()
-        }
-        buttonType={buttonType}
-        buttonColor={buttonColor}
+  return (
+    <ButtonBase
+      buttonColor={buttonColor}
+      buttonType={buttonType}
+      disabled={disabled}
+      isFullWidth={isFullWidth}
+      isLoading={loadingVal}
+      onClick={
+        !disabled && !loadingVal ? onClick : (event) => event.preventDefault()
+      }
+      textColor={textColor}
+      type="button"
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...rest}
+    >
+      <ButtonContents
+        hasIcon={!!icon}
         isLoading={loadingVal}
-        type="button"
-        textColor={textColor}
         isFullWidth={isFullWidth}
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        {...rest}
       >
-        <ButtonContents
-          isLoading={loadingVal}
+        {icon}
+        <ButtonText
           hasIcon={!!icon}
-          isFullWidth={isFullWidth}
-        >
-          {icon}
-          <ButtonText
-            isLoading={loadingVal}
-            hasIcon={!!icon}
-            css={css`
-              padding-top: 2px;
-            `}
-          >
-            {children}
-          </ButtonText>
-        </ButtonContents>
-        <Loader
-          buttonColor={buttonColor}
-          buttonType={buttonType}
-          disabled={disabled}
-          isFullWidth={isFullWidth}
           isLoading={loadingVal}
```

Comment:
[Nit] It looks like we re-arranged some props in alphabetical order, but not entirely: `isFullWidth` after `isLoading` in `ButtonContents`, `css` last instead of first in `ButtonText`.

Reply:
done